### DISCUSSION TITLE
Flow Sensor for Chemistry

### DISCUSSION THREAD

**Discussion by chrisimaxwell** [tagyoureit/nodejs-poolController#898] [Q&A]
Huge Nixie fan posting my first question. But before that: thank you all for the excellent work and documentation. My system is up and running and I am so appreciative.

I have a Nixie dual-body setup with two discrete systems (pool, spa).
I am using Atlas Scientific pH/ORP/temp/pressure sensors and dosing pumps (again, two discrete but identical systems).
I have MQTT, HomeAssistant, and NodeRed all running and integrated.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/6091301/909d6f54-5cf2-4714-9b57-7e9126854c20)

I found flow cells online prohibitively expensive so I built my own from [acrylic blocks](https://www.amazon.com/gp/product/B07PVW7X99) (2 for $100 including the taps and bits).

**THE PROBLEM:** I have not figured out a good way to sense flow for chemical dispensing safety. My current options:

- I have the flow state in MQTT (true/false, via NodeRed) but can't figure out how to integrate this boolean back into the chem controller.  **My main question is, is this possible?**
- I bought cheap [hall-effect flow sensors](https://www.amazon.com/gp/product/B07RH75X2Z) from amazon, but didn't figure out how to integrate them. I have tried and failed. Could keep trying. **Maybe this would be the best route?**
- I considered taking flow data from the INFELLIFLO3 VSF pumps, already in njs-pC, but this doesn't seem to work (yet?).
       (trmb-vast already requested this feature:       https://github.com/tagyoureit/nodejs-poolController/issues/858)

![signal-2024-02-14-200226](https://github.com/tagyoureit/nodejs-poolController/assets/6091301/1df4a35c-6741-46c7-aade-f3bebf37d04c)

So the question is: how should I sense flow.
Thank you again. 
--------------------------

Here are some other pics, just because I appreciate all of the other pics posted here.
HomeAssistant data collection:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/6091301/b98f1d3a-3a63-44b9-bf4f-b236a7fefeb9)

NodeRed integration
![image](https://github.com/tagyoureit/nodejs-poolController/assets/6091301/7f0c1178-dc18-45ae-914e-015b4fdbd187)

A higher-resolution view of the finished acrylic flow cell
![IMG_4800](https://github.com/tagyoureit/nodejs-poolController/assets/6091301/7cdbc974-3160-4a52-b7c2-28c9a285b99d)

[MOSFET relay controller board](https://www.amazon.com/gp/product/B0893MKNB2), to run the relays in my cannibalized Pentair breaker/controller box
![IMG_4807](https://github.com/tagyoureit/nodejs-poolController/assets/6091301/395958a8-efdc-4f4e-b9ac-1f2f2a00504f)

**Comment by tagyoureit** [comment]
Thanks and welcome!  Nice job on the acrylic.  

Check this part of the [wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller#set-up-flow-sensor) out.  If you have any of the Sequent Microsystems cards that read inputs you will be able to use the hall sensor with that.  This [article](https://electrocredible.com/raspberry-pi-pico-flow-sensor-yfs201/) shows how to use a logic level converter to read the output.  That could be a start.

> flow data from the INFELLIFLO3 VSF pumps

These pumps don't use real flow sensors.  As @rstrouse has pointed out a number of times, they do some math to use the watts and rpm to approximate flow.  I wouldn't rely on that as a true measure of flow in your pool.
